src/test/java/lambda/part3/exercise/Exercise2.java

```diff
@@ -0,0 +1,165 @@
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается в список с использованием заданной функции.
+         * ([T], (T -> [R])) -> [R]
+         * @param flatMapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> flatMap(Function<T, List<R>> flatMapping) {
+            List<R> result = new ArrayList<>();
+            getMapped().forEach(element-> result.addAll(flatMapping.apply(element)));
+            return from(result);
+        }
+    }
+
+    @Test
+    void mapEmployeesToLengthOfTheirFullNamesUsingMapHelper() {
+        List<Employee> employees = getEmployees();
+
+        Function<Employee, Person> personExtractor = Employee::getPerson;
+        Function<Person, String> fullNameExtractor = Person::getFullName;
+        Function<String, Integer> stringToLengthConvertor = String::length;
+
+        List<Integer> lengths = null;
+        // TODO                 MapHelper.from(employees)
+        // TODO                          .map(Employee -> Person)
+        // TODO                          .map(Person -> String(full name))
+        // TODO                          .map(String -> Integer(length of string))
+        // TODO                          .getMapped();
+       lengths = MapHelper.from(employees)
+                .map(personExtractor)
+                .map(fullNameExtractor)
+                .map(stringToLengthConvertor)
+                .getMapped();
+        assertThat(lengths, contains(14, 19, 14, 15, 14, 16));
+    }
+
+    private static List<Character> getCharacters(String str) {
+        List<Character> result = new ArrayList<>();
+        for (char ch : str.toCharArray())
+            result.add(ch);
+        return result;
+    }
+
+    @Test
+    void mapEmployeesToCodesOfLetterTheirPositionsUsingMapHelper() {
+        List<Employee> employees = getEmployees();
+        Function<Employee, List<JobHistoryEntry>> jobHistoryExtractor = Employee::getJobHistory;
+        Function<JobHistoryEntry, String> jobPositionExtractor = JobHistoryEntry::getPosition;
+        Function<String, List<Character>> stringToChars =  Exercise2::getCharacters;
+        Function<Character, Integer> charToCode = ch -> (int)ch;
+        List<Integer> codes = null;
+        // TODO               MapHelper.from(employees)
+        // TODO                        .flatMap(Employee -> JobHistoryEntry)
+        // TODO                        .map(JobHistoryEntry -> String(position))
+        // TODO                        .flatMap(String -> Character(letter))
+        // TODO                        .map(Character -> Integer(code letter)
+        // TODO                        .getMapped();
+        codes = MapHelper.from(employees)
+                .flatMap(jobHistoryExtractor)
+                .map(jobPositionExtractor)
+                .flatMap(stringToChars)
+                .map(charToCode)
```

Comment:
I'm not sure that we can use this method at this moment, cos in todo section we have 2 steps here. Just think about it
![screenshot_1](https://user-images.githubusercontent.com/30801520/46312420-b165b380-c5cd-11e8-9884-393bd512c0eb.png)

Reply:
I  checked "todo" section and I figured out that I provided the same number of steps. Anyway, the test was green, so can you provide more details on what I should make better.